cfme/tests/infrastructure/test_infra_tag_filters_combination.py

```diff
@@ -33,19 +34,21 @@ def testing_vis_object(request, a_provider, appliance):
     Returns: class object of certain type
     """
     collection_name, param_class = request.param
-    collection_rest = getattr(appliance.rest_api.collections, collection_name)
-
-    if not test_items:
-        pytest.skip('No content found for test!')
-
-    if collection_name in ['templates', 'vms']:
-        item_type = a_provider.data['provisioning']['catalog_item_type'].lower()
-        for resource in collection_rest:
-            if resource.vendor == item_type:
-                return param_class(name=resource.name, provider=a_provider)
+    if not param_class:
```

Comment:
**REQUIRED**: Just a question here, why don't we require the destination to be passed in as well with an option for it to be None, which will default to all. This saves the special cases of vms/clusters.

Reply:
Updated